website/events/admin.py

```diff
@@ -144,11 +144,6 @@ class EventAdmin(DoNextTranslatedModelAdmin):
         (_("Extra"), {"fields": ("slide", "documents"), "classes": ("collapse",)}),
     )
 
-    def get_form(self, request, obj=None, change=False, **kwargs):
-        form = super().get_form(request, obj, change, **kwargs)
```

Comment:
Would changes this not reverse the fix @pingiun tried to implement in https://github.com/svthalia/concrexit/pull/1453?

Reply:
In the summary of that PR he mentioned:
> This commit fixes #1307 by not saving anything in the init of the event model, as this seems to break deleting things from foreign keys.

so I assumed that the problem was that we were saving fields in the `__init__` method of `Event`.
If the actual problem was that the `clean` method in the model shouldn't do these checks then you're probably right.
I could try if my approach to checking if you're allowed to change the price/start time would also work in the `clean_changes` method.

Reply:
> If the actual problem was that the clean method in the model shouldn't do these checks then you're probably right.

Conceptually this is the wrong place to check changes, because models should only check their own integrity and not check if changes are correct. 

So if the test I created doesn't fail this isn't technically a wrong solution but it's uglier because you're doing a database lookup in a `clean()` method. 

I think it's best to check changes in the form clean method rather than the model clean